### PR TITLE
feat : Update thedevelop-userTypeFix branch with the latest features from develop

### LINKED ISSUE #1987: Update the`develop-userTypeFix` branch with the latest features from `develop` - Deleting Recurring Events
**Is your feature request related to a problem? Please describe.**
1. We recently fixed the usage of `userType` with this PR.
    1. https://github.com/PalisadoesFoundation/talawa-api/pull/1830
1. A new `develop-userTypeFix` branch was created to add this feature
1. We need to make sure that the `develop-userTypeFix` branch:
    1. Maintains its current functionality
    1. Includes any new features in the `develop` branch and adjusts them to the new `userType` functionality. 
     1. Has all valid and passing tests that maintain our current test code coverage levels

This is required to support these new features added to `develop`:

1. https://github.com/PalisadoesFoundation/talawa-api/pull/1890
1. https://github.com/PalisadoesFoundation/talawa-api/pull/1952
1. https://github.com/PalisadoesFoundation/talawa-api/pull/1917
1. https://github.com/PalisadoesFoundation/talawa-api/pull/1960

**Describe the solution you'd like**
1. Merge `develop` into the `develop-userTypeFix` branch
3. Make sure all newly added features to the `develop` branch are included in the  `develop-userTypeFix` branch.
4. The new functionality of `userType` must be unchanged.
5. No other features must be affected
6. All tests must be valid and pass

**Describe alternatives you've considered**

- N/A

**Approach to be followed (optional)**

- See above

**Additional context**
1. This is going to require someone with experience with the repo, or proven long term experience with typescript, MERN, and / or graphQL in their GitHub profile

Research these issues for context with the new `userType` functionality:

1. https://github.com/PalisadoesFoundation/talawa-api/issues/1965

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship 

- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Atharva-Kanherkar**
**What kind of change does this PR introduce?**

 Feature

**Issue Number:**

Fixes #1987 

**Did you add tests for your changes?**

 Yes

 

**Summary**

Merging the develop branch. 

**Does this PR introduce a breaking change?**
No

 

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

 Yes

### CODE REVIEW

**Comment by Atharva-Kanherkar** [comment]
@AVtheking Please whenever you are free review the tests directory, many of them are failing, the test code coverage is very low due to unkown reasons. I would really love some help so that we can get this merged as soon as possible.

**Comment by palisadoes** [comment]
Please fix the linting. The tests are not running.
- https://github.com/PalisadoesFoundation/talawa-api/actions/runs/8349581390/job/22854067087?pr=2032

File: src/utilities/isAuthCheck.ts
Comment (AVtheking) [inline src/utilities/isAuthCheck.ts:1]: Remove the userType here.

**Comment by AVtheking** [comment]
@Atharva-Kanherkar you still have used userType field in many places may be that's why testas are failing remove those.

**Comment by palisadoes** [comment]
@Atharva-Kanherkar 

There is still some work to do.

1. You still have references to userType in the `src/` directory. This is an improvement, 36 instances vs 100 in the original `develop-userTypeFix` branch

    image

1. There are 533 references to userType in the `tests/` directory vs 620 in the original `develop-userTypeFix` branch. 

    image

It's a lot of outstanding work. Do you think you'll be able to finish by the weekend?

**Comment by palisadoes** [comment]
@Atharva-Kanherkar 

1. Take a look here. I think I know what the problem is. I was wondering why the tests are posing such a challenge. You'll need to coordinate with the others to submit tests. 
    1. https://github.com/PalisadoesFoundation/talawa-api/issues/1965#issuecomment-2009351901
1. I think the best approach is to:
    1.  fix the last few references to userType in `src/` in this PR
    2. get the others to assist in fixing the tests as a team

You'll need to coordinate that yourselves through slack as you are all in the same timezone.

**Comment by AVtheking** [comment]
@Atharva-Kanherkar you have merged the new features in develop branch directly but you also need to refactor them according to this branch along with their tests, that was the issue you have to fix, ask me if you need any help.

**Comment by palisadoes** [comment]
At this stage we need as many people to pitch in.

Is there a way for others to make commits to an existing PR?

**Comment by varshith257** [comment]
@palisadoes Yes, there is way . @Atharva-Kanherkar  in their forked repository has to add collaborators who can directly push changes to that branch with write access. This way, multiple people can work collaboratively on the changes required.

image

For example I have invited you as collaborator for my forked repository

**Comment by palisadoes** [comment]
1. @Atharva-Kanherkar Can you do that to speed things up?
2. @varshith257 Would that mean the commits to the the forked branch by the collaborators would then be immediately visible here?

**Comment by Atharva-Kanherkar** [comment]
@palisadoes I've been traveling to my home and I'll reach in two hours. Sorry for the delay. Will do that as soon as I reach home.

**Comment by AVtheking** [comment]
> 1. @Atharva-Kanherkar Can you do that to speed things up?
> 
>     2. @varshith257 Would that mean the commits to the the forked branch by the collaborators would then be immediately visible here?

Yes it would be visible but I think there is no need to fork after making contributor we can directly clone the repo and do commits.

**Comment by varshith257** [comment]
> 2. @varshith257 Would that mean the commits to the the forked branch by the collaborators would then be immediately visible here?

@palisadoes Yes, once changes are pushed to the branch associated with the existing PR in the forked repository, changes will automatically reflect in the original PR  because they are making the changes to that branch.

**Comment by palisadoes** [comment]
1. This solution looks like the best option. 
2. We need to get this merged before the weekend when most contributors will become active again.
3. How would we split the responsibilities in a way that there is no overlapping work?

**Comment by palisadoes** [comment]
@Atharva-Kanherkar Let us know when you have added collaborators

**Comment by Atharva-Kanherkar** [comment]
@palisadoes  @AVtheking  @varshith257  I have added you guys to the collaborators. 
image

**Comment by Atharva-Kanherkar** [comment]
> @Atharva-Kanherkar
> 
> There is still some work to do.
> 
> 1. You still have references to userType in the `src/` directory. This is an improvement, 36 instances vs 100 in the original `develop-userTypeFix` branch
>    image
> 2. There are 533 references to userType in the `tests/` directory vs 620 in the original `develop-userTypeFix` branch.
>    image
> 
> It's a lot of outstanding work. Do you think you'll be able to finish by the weekend?

I can try. I have just reached home. Will get to work.

**Comment by palisadoes** [comment]
We are coordinating in slack here:
- https://thepalisadoes-dyb6419.slack.com/archives/C06NRV8J9A9/p1710952232280639

**Comment by AVtheking** [comment]
@Atharva-Kanherkar  thanks for adding me as collaborator you try to fix some files from the spreadsheet ,tomorrow I have exam ,I have to study for it so I will be working on it after my exam.

**Comment by varshith257** [comment]
@Atharva-Kanherkar Also add @adi790uu and @kanhaiya04

**Comment by kanhaiya04** [comment]
Yes, @Atharva-Kanherkar add me as well.

**Comment by varshith257** [comment]
@Atharva-Kanherkar Any update. Let us know after adding all the colloborators and find us on slack channel mentioned above. So that we can coordinate easily

**Comment by Atharva-Kanherkar** [comment]
> @Atharva-Kanherkar Any update. Let us know after adding all the colloborators and find us on slack channel mentioned above. So that we can coordinate easily

I have added all of  you guys.

File: src/utilities/isAuthCheck.ts
Comment (Atharva-Kanherkar) [reply src/utilities/isAuthCheck.ts:1]: How do i make sure the user is an admin as we do not have idea about which organization to check for in the appuserprofile?

**Comment by Atharva-Kanherkar** [comment]
I have added my name in the spreadsheet on the files i am working on. @palisadoes @AVtheking

**Comment by palisadoes** [comment]
The spreadsheet is ready. Only one person has signed up.

**Comment by palisadoes** [comment]
Please check slack on how we need to handle the retirement of `userType`
- https://thepalisadoes-dyb6419.slack.com/archives/C06NRV8J9A9/p1711031414872449

**Comment by palisadoes** [comment]
1. Check the PR workflows here when you make your updates in the fork. 
2. Linting is failing now

File: tests/resolvers/AgendaItem/categories.spec.ts
Comment (varshith257) [inline tests/resolvers/AgendaItem/categories.spec.ts:44]: @Atharva-Kanherkar I think we need to modify tests according to the AppUserProfile instead of deletion. It leads to low test coverage

Comment (Atharva-Kanherkar) [reply tests/resolvers/AgendaItem/categories.spec.ts:44]: Yes, i am trying to do exactly that. Thank you

**Comment by Atharva-Kanherkar** [comment]
@AVtheking why do you think still the tests are failing? The venue feature needs to be refactored?

**Comment by AVtheking** [comment]
@Atharva-Kanherkar  yes venue need to be refactored.

**Comment by Atharva-Kanherkar** [comment]
@palisadoes  I have ensured 100% test code coverage and all tests pass in the agenda item files.

**Comment by varshith257** [comment]
@AVtheking @Atharva-Kanherkar Check the latest workflow test of Testing Application(20.x) PR. All failing tests are exposed in this branch.

[PR Workflow](https://github.com/PalisadoesFoundation/talawa-api/actions/workflows/pull-request.yml)

**Comment by varshith257** [comment]
I have added `.env_test `file in my code and triggered the action though it's not capturing.

 @Anubhav-2003 Can check out the failing test of  `.env_test`. And why it's not able to capture  .env_test added in .gitignore?

**Comment by Anubhav-2003** [comment]
> I have added `.env_test `file in my code and triggered the action though it's not capturing.
> 
> @Anubhav-2003 Can check out the failing test of `.env_test`. And why it's not able to capture .env_test added in .gitignore? It's related to your work done in develop branch.

this is because you people have implemented a faulty version of the setup. If you see the current refactored version of the set up has around 900 lines. You people are still using the old version of the setup.ts. Also you people have removed the line where the .env_test file would be automatically created inside the set up. Please fix your set up file.

**Comment by varshith257** [comment]
We haven't looked at that file. We are joined with @Atharva-Kanherkar to speedup the work recently. Ok , we will make sure to refactor it

**Comment by Anubhav-2003** [comment]
> We haven't looked at that file. We are joined with @Atharva-Kanherkar to speedup the work recently. Ok , we will make sure to refactor it

No problem, collaborative work is always appreciated, if you have any more doubts,  or if any help is needed from my side. Please let me know. Thank you.

**Comment by palisadoes** [comment]
@Anubhav-2003 

1. Join the #proj-user-role slack channel
2. Feel free to join in and refactor `setup.ts` according to this requirement. @Atharva-Kanherkar please add @Anubhav-2003 as a collaborator to your repo

**Comment by Atharva-Kanherkar** [comment]
> @palisadoes sure sir, also I can assist not just in setup but in other parts as well, if the present contributors feel they need any assistance. When this PR will be merged with develop, a few bugs may surface initially. I will be happy to fix them through separate PRs. Thank you.

I hace added you as a contributor. Seperate PRs will not make any sense for this issue, I guess it would just add up to the work as we have freezed the merging of other PRs, please help in this to us, it would be really helpful.

**Comment by Anubhav-2003** [comment]
> > @palisadoes sure sir, also I can assist not just in setup but in other parts as well, if the present contributors feel they need any assistance. When this PR will be merged with develop, a few bugs may surface initially. I will be happy to fix them through separate PRs. Thank you.
> 
> I hace added you as a contributor. Seperate PRs will not make any sense for this issue, I guess it would just add up to the work as we have freezed the merging of other PRs, please help in this to us, it would be really helpful.



> > @palisadoes sure sir, also I can assist not just in setup but in other parts as well, if the present contributors feel they need any assistance. When this PR will be merged with develop, a few bugs may surface initially. I will be happy to fix them through separate PRs. Thank you.
> 
> I hace added you as a contributor. Seperate PRs will not make any sense for this issue, I guess it would just add up to the work as we have freezed the merging of other PRs, please help in this to us, it would be really helpful.

What you are saying makes no sense. once you merge this PR into the develop, there will be consequences in the other repositories as well because the merging for this with the current develop has not yet been tested for the entire talawa suite. Just like the setup file, there will be other files whose errors might not initially pop up in the workflow. But there will be instances of inconsistent merging for fixing those we would need separate PRs, I hope you understand what I’m saying. Thank you

**Comment by AVtheking** [comment]
> > > @palisadoes sure sir, also I can assist not just in setup but in other parts as well, if the present contributors feel they need any assistance. When this PR will be merged with develop, a few bugs may surface initially. I will be happy to fix them through separate PRs. Thank you.
> > 
> > 
> > I hace added you as a contributor. Seperate PRs will not make any sense for this issue, I guess it would just add up to the work as we have freezed the merging of other PRs, please help in this to us, it would be really helpful.
> 
> > > @palisadoes sure sir, also I can assist not just in setup but in other parts as well, if the present contributors feel they need any assistance. When this PR will be merged with develop, a few bugs may surface initially. I will be happy to fix them through separate PRs. Thank you.
> > 
> > 
> > I hace added you as a contributor. Seperate PRs will not make any sense for this issue, I guess it would just add up to the work as we have freezed the merging of other PRs, please help in this to us, it would be really helpful.
> 
> What you are saying makes no sense. once you merge this PR into the develop, there will be consequences in the other repositories as well because the merging for this with the current develop has not yet been tested for the entire talawa suite. Just like the setup file, there will be other files whose errors might not initially pop up in the workflow. But there will be instances of inconsistent merging. For fixing those we would need separate PRs, I hope you understand what I’m saying. Thank you

yes after merging it in develop we need to lookout for any breaking thing and resolves them as soon as possible.

**Comment by varshith257** [comment]
@Anubhav-2003 Cool! Once it got merged the bugs could be there and if you found then make a PR against a parent issue to track

**Comment by Anubhav-2003** [comment]
> @Anubhav-2003 Cool! Once it got merged the bugs could be there and if you found then make a PR against a parent issue to track

Sure I will

**Review by varshith257 — APPROVED** [review]
LGTM

**Pull request merged**
